server/db/errors.go

```diff
@@ -40,11 +42,13 @@ func (ae ArchiveError) Error() string {
 // SameErrorTypes checks for error equality or ArchiveError.Code equality if
 // both errors are of type ArchiveError.
 func SameErrorTypes(errA, errB error) bool {
-	if errA == errB {
+	if errors.Is(errA, errB) {
```

Comment:
Are we sure that this won't return true for two `ArchiveError`s with different codes? I haven't used `errors.Is` until now.  Perhaps a test for `SameErrorTypes`?